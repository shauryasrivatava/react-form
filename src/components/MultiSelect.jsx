import React from 'react';
import PropTypes from 'prop-types';

const MultiSelect = (props) => {
    return (
        <div>
            <label htmlFor={props.labelFor}>{props.label}</label>
            <select multiple className={props.className} value={props.value} id={props.labelFor} name={props.labelFor} onChange={props.handleChange}>
                {props.children}
            </select>
        </div>
    )
}

MultiSelect.propTypes = {
    className: PropTypes.string,
    labelFor: PropTypes.string,
    value: PropTypes.array,
    handleChange: PropTypes.func,
    label: PropTypes.string
};

MultiSelect.defaultProps = {
    className: 'form-group',
    labelFor: 'checkbox',
    value: [],
    handleChange: function(){},
    label: 'MultiSelect'
}

export default MultiSelect;