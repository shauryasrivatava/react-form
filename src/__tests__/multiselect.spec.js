import renderer from 'react-test-renderer';
import MultiSelect from '../components/MultiSelect';
import React from 'react';
import { shallow } from 'enzyme';
import enzymeConfig from '../../enzymeConfig';

describe('<MultiSelect />', function(){

    it('Should render correctly', function() {

        const component = renderer.create(<MultiSelect />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    })

    it('Should have the correct label', function() {

        const component = shallow(<MultiSelect label="hello world"/>);
        expect(component.find('label').text()).toEqual('hello world');

    })

    it('Should have one <select>', function() {

        const component = shallow(<MultiSelect />);
        expect(component.find('select').length).toBe(1);

    })

})