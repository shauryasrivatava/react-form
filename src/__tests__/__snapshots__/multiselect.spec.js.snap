// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultiSelect /> Should render correctly 1`] = `
<div>
  <label
    htmlFor="checkbox"
  >
    MultiSelect
  </label>
  <select
    className="form-group"
    id="checkbox"
    multiple={true}
    name="checkbox"
    onChange={[Function]}
    value={Array []}
  />
</div>
`;
